### PR TITLE
server: add creating raft cluster when become leader.

### PR DESCRIPTION
**Pull request by qiuyesuifeng**

### CODE REVIEW

**Comment by siddontang** [comment]
No check in getRaftCluster?

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @ngaut  @disksing

File: server/cluster.go [hunk 1]
```diff
@@ -139,25 +139,32 @@ func (s *Server) getRaftCluster() (*raftCluster, error) {
...
+	}
+
 	value, err := getValue(s.client, s.getClusterRootPath())
 	if err != nil {
-		return nil, errors.Trace(err)
+		return errors.Trace(err)
 	}
 	if value == nil {
-		return nil, nil
+		return nil
```
Comment (disksing) [inline server/cluster.go]: Should we return an error here?

Comment (qiuyesuifeng) [inline server/cluster.go]: Seems not, if we have not bootstrapped the cluster, then the cluster value from etcd will be nil.
@siddontang  Am i right?

Comment (siddontang) [inline server/cluster.go]: yes, the raft cluster is nil if not bootstrapped.

**Comment by disksing** [comment]
LGTM.

**Pull request merged**